src/pvrclient-nextpvr.cpp

```diff
@@ -1734,13 +1752,18 @@ PVR_ERROR cPVRClientNextPVR::UpdateTimer(const PVR_TIMER &timerinfo)
 
 /************************************************************/
 /** Live stream handling */
-
 bool cPVRClientNextPVR::OpenLiveStream(const PVR_CHANNEL &channelinfo)
 {
+  if (channelinfo.iUniqueId == m_iLiveStreamUID)
+  {
+    XBMC->Log(LOG_NOTICE, "New channel uid equal to the already streaming channel. Skip re-tune.");
+    return true;
+  }
+  
   m_PlaybackURL = "";
+  m_iLiveStreamUID = 0;
 
   XBMC->Log(LOG_DEBUG, "OpenLiveStream(%d:%s) (oid=%d)", channelinfo.iChannelNumber, channelinfo.strChannelName, channelinfo.iUniqueId);
-  if (strstr(channelinfo.strStreamURL, "live?channel") == NULL)
   {
```

Comment:
If you remove the condition, please also remove the corresponding block markers (the curly braces).